lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -67,6 +67,10 @@ public function validateMapping()
         $cmf = $this->em->getMetadataFactory();
         $classes = $cmf->getAllMetadata();
 
+        if (count($classes) === 0) {
+            return false;
```

Comment:
1. return type is inconsistent with signature
2. an empty set of mappings is perfectly valid here